instrumentation/spring/spring-integration-4.1/javaagent/spring-integration-4.1-javaagent.gradle

```diff
@@ -0,0 +1,50 @@
+apply from: "$rootDir/gradle/instrumentation.gradle"
+
+ext {
+  // context "leak" here is intentional: spring-integration instrumentation will always override
+  // "local" span context with one extracted from the incoming message
+  doNotFailOnContextLeak = true
+}
```

Comment:
This is a consequence of always using the context extracted from the incoming message; we completely ignore the "local" current context, but that's fine in that case.

Reply:
Hmm... our `debugContextLeakIfEnabled` should not be called at all in case of `CONSUMER` spans. This is not spring integration specific problem.

Reply:
It's called for all spans, span kind does not matter.

Reply:
Yes, but the logic of not failing the test should be the same for all `CONSUMER` spans.